tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py

```diff
@@ -0,0 +1,29 @@
+from unittest import TestCase
+from parameterized import parameterized
+
+from samcli.hook_packages.terraform.hooks.prepare.resource_linking import _get_configuration_address
+
+
+class TestResourceLinking(TestCase):
+    @parameterized.expand(
+        [
+            (
+                "module.get_language_function.aws_lambda_function.this[0]",
+                "module.get_language_function.aws_lambda_function.this",
+            ),
+            (
+                "module.get_language_function.aws_lambda_function.this[1]",
+                "module.get_language_function.aws_lambda_function.this",
+            ),
+            ("module.functions[0].aws_lambda_function.this[0]", "module.functions.aws_lambda_function.this"),
+            ("module.functions[1].aws_lambda_function.this[1]", "module.functions.aws_lambda_function.this"),
+            (
+                'module.functions["get_function"].aws_lambda_function.this[0]',
+                "module.functions.aws_lambda_function.this",
+            ),
+        ]
+    )
+    def test_get_configation_address(self, input_addr, expected_addr):
```

Comment:
can you add a test case if we passed a string without any indices. something like `module.functions.aws_lambda_function.this`